### DISCUSSION TITLE
TP-Link Vigi C330i snapshot not working

### DISCUSSION THREAD

**Discussion by wcgoh** [koush/scrypted#1027] [Q&A]
Able to use ONVIF to stream TP-link Vigi C330i (Wired Camera) but faced issue with generating snapshot. Here is the error codes

`listening events

[HomeKit]: HomeKit Connection 10.0.0.77
[Snapshot Plugin]: Snapshot failed Error: ffmpeg input to image conversion failed with exit code: 1, /usr/bin/ffmpeg -hide_banner -y -f image2pipe -i pipe:4 -filter_complex scale=-2:'min(720,ih)' -frames:v 1 -f image2 pipe:3

[Snapshot Plugin]:     at /src/ffmpeg-image-filter.ts:178:23
[Snapshot Plugin]:     at processTicksAndRejections (node:internal/process/task_queues:95:5)
[Snapshot Plugin]: creating error image with background Snapshot Failed true
[HomeKit]: snapshot error RPCResultError: ffmpeg input to image conversion failed with exit code: 1, /usr/bin/ffmpeg -hide_banner -y -f image2pipe -i pipe:4 -filter_complex eq=brightness=-0.2[snapshotBlur] ; [snapshotBlur] gblur=sigma=25[snapshotText] ; [snapshotText] drawtext=fontfile=/server/volume/plugins/@scrypted/snapshot/zip/unzipped/fs/Lato-Bold.ttf:text='Snapshot Failed':fontcolor=white:fontsize=h/8:x=(w-text_w)/2:y=(h-text_h)/2 -frames:v 1 -f image2 pipe:3
[HomeKit]:     at /src/ffmpeg-image-filter.ts:178:23
[HomeKit]:     at processTicksAndRejections (node:internal/process/task_queues:95:5)
[HomeKit]: @scrypted/snapshot:host
[HomeKit]: host:@scrypted/homekit {
[HomeKit]:   cause: undefined
[HomeKit]: }
[Snapshot Plugin]: Snapshot failed Error: ffmpeg input to image conversion failed with exit code: 1, /usr/bin/ffmpeg -hide_banner -y -f image2pipe -i pipe:4 -filter_complex scale=-2:'min(720,ih)' -frames:v 1 -f image2 pipe:3
[Snapshot Plugin]:     at /src/ffmpeg-image-filter.ts:178:23
[Snapshot Plugin]:     at processTicksAndRejections (node:internal/process/task_queues:95:5)
[Snapshot Plugin]: creating error image with background Snapshot Failed true
[Snapshot Plugin]: Snapshot failed Error: ffmpeg input to image conversion failed with exit code: 1, /usr/bin/ffmpeg -hide_banner -y -f image2pipe -i pipe:4 -filter_complex scale=-2:'min(720,ih)' -frames:v 1 -f image2 pipe:3
[Snapshot Plugin]:     at /src/ffmpeg-image-filter.ts:178:23
[Snapshot Plugin]:     at processTicksAndRejections (node:internal/process/task_queues:95:5)
[Snapshot Plugin]: creating error image with background Snapshot Failed true
[Snapshot Plugin]: Snapshot failed Error: ffmpeg input to image conversion failed with exit code: 1, /usr/bin/ffmpeg -hide_banner -y -f image2pipe -i pipe:4 -filter_complex scale=-2:'min(720,ih)' -frames:v 1 -f image2 pipe:3
[Snapshot Plugin]:     at /src/ffmpeg-image-filter.ts:178:23
[Snapshot Plugin]:     at processTicksAndRejections (node:internal/process/task_queues:95:5)
[Snapshot Plugin]: creating error image with background Snapshot Failed true
`

**Comment by koush** [comment]
does it work in management console?

**Comment by wcgoh** [comment]
> does it work in management console?

Yes it works. If I click on the failed snapshot, it can stream.
